cpu/esp32/esp_xtimer.c

```diff
@@ -93,10 +93,9 @@ void _ets_to_xtimer_free(ETSTimer *timer)
 /* xtimer call back function, distributes ets_timer callbacks */
 void IRAM_ATTR _ets_to_xtimer_callback (void *arg)
 {
+    CHECK_PARAM(arg != NULL);
     struct _ets_to_xtimer* e2xt = (struct _ets_to_xtimer*)arg;
-
-    CHECK_PARAM (e2xt != NULL);
-    CHECK_PARAM (e2xt->ets_timer != NULL);
+    CHECK_PARAM(e2xt->ets_timer != NULL);
```

Comment:
To be honest, the code seems less readable with this order. I used `CHECK_PARAM` here to avoid a bunch of `if` blocks with `DEBUG` and returns and to keep the code more readable.

The question is whether `CHECK_PARAM` should protect against soft or hard errors. For soft errors, the check with a return would be correct. For hard errors, an `assert` would be better.

Would cpp-check be happy with `assert`s in original order?